src/main/java/ch/njol/skript/expressions/ExprAmount.java

```diff
@@ -109,6 +124,48 @@ protected Long[] get(Event e) {
 		return new Long[]{(long) exprs.getArray(e).length};
 	}
 
+	@Override
+	public @Nullable Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (any != null) {
+			return switch (mode) {
+				case SET, ADD, RESET, DELETE, REMOVE -> CollectionUtils.array(Number.class);
+				default -> null;
+			};
+		}
+		return super.acceptChange(mode);
+	}
+
+	@Override
+	public void change(Event event,  Object@Nullable[] delta, Changer.ChangeMode mode) {
+		if (any == null) {
+			super.change(event, delta, mode);
+			return;
+		}
+		double amount = delta != null ? ((Number) delta[0]).doubleValue() : 0;
```

Comment:
You could use `1` here instead of `0`, and then you would not need the `amount = 1` fall-through case for RESET and DELETE